### PR TITLE
Grid update

### PR DESCRIPTION
**Pull request by Nilanshrajput**
## Description

Update SyferText for Pygrid compatibility
Got tutorials running on grid nodes,
add protocol commands for encrypted vectors

## Affected Dependencies
Black version updates to 20.8b1 and Pysyft to 0.2.9

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: syfertext/doc.py [hunk 1]
```diff
@@ -221,8 +220,8 @@ def get_vector(self, excluded_tokens: Dict[str, Set[object]] = None):
         """Get document vector as an average of in-vocabulary token's vectors,
         excluding token according to the excluded_tokens dictionary.
 
-        Args:
-            excluded_tokens (Dict): A dictionary used to ignore tokens of the document based on their custom attribute values
+        Args
+            excluded_tokens (Dict): A dictionary used to ignore tokens of the document based on values
```
Comment (Nilanshrajput) [inline syfertext/doc.py:224]:
```suggestion
            excluded_tokens (Dict): A dictionary used to ignore tokens of the document based on their custom attribute values
```

File: setup.py [hunk 2]
```diff
@@ -23,12 +23,12 @@ def read(fname):
     install_requires=[
         "tqdm==4.36.1",
         "mmh3==2.5.1",
-        "syft==0.2.8",
+        "syft @ git+https://github.com/OpenMined/PySyft.git@v0.2.9#egg=syft",
         "requests==2.22.0",
```
Comment (AlanAboudib) [inline setup.py:27]: can you please add your fork here that has PyGrid serialization correction

File: syfertext/doc.py [hunk 3]
```diff
@@ -221,7 +220,7 @@ def get_vector(self, excluded_tokens: Dict[str, Set[object]] = None):
         """Get document vector as an average of in-vocabulary token's vectors,
         excluding token according to the excluded_tokens dictionary.
 
-        Args:
+        Args
```
Comment (AlanAboudib) [inline syfertext/doc.py:223]: Please add the colon

File: syfertext/doc.py [hunk 4]
```diff
@@ -354,30 +360,14 @@ def get_encrypted_token_vectors(
...
-            *workers, crypto_provider=crypto_provider, requires_grad=requires_grad
+            *workers,
+            crypto_provider=crypto_provider,
+            requires_grad=requires_grad,
+            protocol=protocol,
         )
 
         return token_vectors
 
-    def decode_logits(
```
Comment (AlanAboudib) [inline syfertext/doc.py:362]: Why the decode logits method is removed?

File: syfertext/language.py [hunk 5]
```diff
@@ -501,6 +507,10 @@ def __call__(self, text: Union[str, String, StringPointer]) -> Union[Doc, DocPoi
         # or the Doc to be processed is located
         if isinstance(text, ObjectPointer):
             location_id = text.location.id
+
+            # deploy all the states of Pipeline on location of text
+            if location_id not in self.deployed_on:
```
Comment (AlanAboudib) [inline syfertext/language.py:512]: As we discussed, deployment should be done in the deploy() method only.

**Review by AlanAboudib — CHANGES_REQUESTED** [review]
Here is my review. Would it be possible to work on it asap? I will build on top of it today once it is merged

File: syfertext/doc.py (hunk 4 above)
Comment (Nilanshrajput) [reply syfertext/doc.py]: I don't remember removing it, I removed by mistake

File: syfertext/doc.py [hunk 6]
```diff
@@ -368,6 +369,25 @@ def get_encrypted_token_vectors(
...
+        # If 'mpc' encryption is used, decrypt the logits
+        if encryption == "mpc":
+            logits = logits.get().float_precision()
+
+        # Get the predict label text
+        label = labels[logits.argmax().item()]
+
+        # Set a custom attribute to the Doc containing the predicted label
+        self.set_attribute(name=task_name, value=label)
+
```
Comment (AlanAboudib) [inline syfertext/doc.py]:
```suggestion
    def decode_logits(
        self,
        task_name: str,
        logits: AbstractTensor,
        labels: List[str],
        single_label: bool,
        encryption: str,
    ) -> None:
        """Decodes the logits tensor produced by a classifier
        and gets the predicted label. Then this label is set as
        and custom attribute value of the Doc.
        Args:
            task_name: The name of the classification task. This
                will be used as the custom attribute's name.
            logits: This is the logits tensor.
            labels: The labels textual names of the classification
                task.
            single_label: A flag to distinguish the number of labels
                to predict.
            encryption: The encryption scheme used. For example, 'mpc'.
        Todo:
            For the moment, only single label classifier logits
                are supported, this should be later extended to
                multi-label classifiers.
        """

        # If 'mpc' encryption is used, decrypt the logits
        if encryption == "mpc":
            logits = logits.get().float_precision()

        # Get the predict label text
        label = labels[logits.argmax().item()]

        # Set a custom attribute to the Doc containing the predicted label
        self.set_attribute(name=task_name, value=label)

```

**Review by AlanAboudib — CHANGES_REQUESTED** [review]
